webapp/app.py

```diff
@@ -215,6 +216,39 @@ def start_career():
     return flask.render_template("/careers/start.html")
 
 
+@app.route("/careers", methods=["GET"])
+def all_careers():
+    context = {
+        "all_departments": _group_by_department(greenhouse.get_vacancies())
+    }
+    context["vacancies"] = greenhouse.get_vacancies()
+    context["vacancies_json"] = [
+        vacancy.to_dict() for vacancy in context["vacancies"]
+    ]
+
+    return flask.render_template("/careers/index.html", **context)
+
+
+@app.route("/careers/all", methods=["GET"])
+def all_jobs():
+    context = {
+        "all_departments": _group_by_department(greenhouse.get_vacancies())
+    }
+    context["vacancies"] = greenhouse.get_vacancies()
+
+    context["vacancies_json"] = [
+        vacancy.to_dict() for vacancy in context["vacancies"]
+    ]
+
+    jobs_list = context["vacancies_json"]
+
+    return flask.render_template(
+        "/careers/all.html",
+        **context,
+        jobs_list=jobs_list,
+    )
```

Comment:
Some of this seems unnecessary.

I can't see any reference in the templates to `jobs_list`, so I don't *think* you need to pass it through to the template. Correct me if I'm wrong.

And then I don't see what's stopping you from instantiating `all_departments` and `vacancies_json` as you pass them through to the template. That way you avoid the potentially confusing `**context` syntax.

Also, is there a reason you are doing `to_dict` rather than using `json.dumps()`? I *think* it should do the same thing.

``` python3
# At the top
import json
```

Then:

```suggestion
    vacancies = greenhouse.get_vacancies()

    return flask.render_template(
        "/careers/all.html",
        all_departments=_group_by_department(greenhouse.get_vacancies()),
        vacancies=vacancies,
        vacancies_json=json.dumps(vacancies)
    )
```